fuel-core/src/schema/tx/types.rs

```diff
@@ -214,8 +216,18 @@ impl Transaction {
 
     async fn status(&self, ctx: &Context<'_>) -> async_graphql::Result<Option<TransactionStatus>> {
         let db = ctx.data_unchecked::<Database>();
-        let status = db.get_tx_status(&self.0.id())?;
-        Ok(status.map(Into::into))
+
+        let txpool = ctx.data::<Arc<TxPool>>().unwrap().pool();
+
+        let transaction_in_pool = txpool.find_one(&self.0.id()).await;
+
+        if transaction_in_pool.is_some() && db.get_tx_status(&self.0.id()).is_err() {
```

Comment:
I think its a safe assumption that we don't need to check the db if it was found in the pool. 

```suggestion
        if transaction_in_pool.is_some() {
```

Reply:
Just tested and this check is redundant, fixed it

Reply:
I put this 2nd check in there as I was running into an issue of `Submitted {}` being overreported, looking into it now it seems the root cause is that `submit_tx` isn't clearing transactions from the mempool correctly. Looking into why